### ISSUE TITLE
container provisioning stuck with waiting status...

### ISSUE THREAD

**Issue by rafaeltuelho** [jbossdemocentral/bpms-fuse-travel-agency-integration-demo#13]
The `bookingflightcon` container do not starts. It get stuck with `waiting` status.

Nothing (error or exception) appears on conatiners's fuse/karaf.log 
I tried to stop/start using `--force` flag but no way. Build and deploy the projects again with no success too.

**Comment by weimeilin79** [comment]
Need more information like logs too see what's wrong. Is your other container hotelbooking working?

**Comment by rafaeltuelho** [comment]
There are two containers in this state (`waiting`): `bookingflightcon` and `bookinhotelgcon`. All others starts fine:

```
[id]                [version]  [type]  [connected]  [profiles]                             [blueprint]  [spring]  [provision status]
root*               1.0        karaf   yes          fabric                                 started                success           
                                                    fabric-ensemble-0000-1                                                          
                                                    jboss-fuse-full                                                                 
  bookingflightcon  1.0        karaf   yes          demo-travelagency-bookingservice       waiting                waiting           
  bookinhotelgcon   1.0        karaf   yes          demo-travelagency-hotelbookingservice  waiting                waiting           
  promoflightcon    1.0        karaf   yes          demo-travelagency-promotionflight      started                success           
  promohotelcon     1.0        karaf   yes          demo-travelagency-promotionhotel       started                success           
  wsflightcon       1.0        karaf   yes          demo-travelagency-webendpoint          started                success           
  wshotelcon        1.0        karaf   yes          demo-travelagency-hotelwsendpoint      started                success
```

As I mentioned there are no problems information on conatiner's log. See a log snippet from `bookingflightcon`:

```
rsoares@rsoares ~/dev/github/jbossdemocentral/bpms-fuse-travel-agency-integration-demo/target/jboss-fuse-6.2.0.redhat-133 (master) $
> tail -F instances/bookingflightcon/data/log/karaf.log

2015-08-31 23:32:21,609 | INFO  | FelixStartLevel  | Activator                        | 117 - org.apache.camel.camel-core - 2.15.1.redhat-620133 | Camel activator starting
2015-08-31 23:32:21,656 | INFO  | FelixStartLevel  | Activator                        | 117 - org.apache.camel.camel-core - 2.15.1.redhat-620133 | Camel activator started
2015-08-31 23:32:21,846 | WARN  | FelixStartLevel  | NamespaceManager                 | 198 - org.springframework.osgi.extender - 1.2.1 | Bundle Apache XBean :: Classloader (org.apache.xbean.classloader) cannot see class [org.springframework.beans.factory.xml.NamespaceHandlerResolver]; ignoring its namespace handlers
2015-08-31 23:32:21,861 | INFO  | FelixStartLevel  | BlueprintContainerImpl           | 14 - org.apache.aries.blueprint.core - 1.4.2 | Bundle travelagency is waiting for namespace handlers [http://aries.apache.org/xmlns/jpa/v1.0.0]
2015-08-31 23:32:49,969 | WARN  | a01c1-1-thread-1 | AgentUtils                       | 83 - io.fabric8.fabric-agent - 1.2.0.redhat-133 | Could not get httpUrl from fabricService
2015-08-31 23:32:50,297 | WARN  | nload-2-thread-1 | JaxbUtil                         | 83 - io.fabric8.fabric-agent - 1.2.0.redhat-133 | Old style feature file without namespace found (URI: mvn:org.apache.activemq/activemq-karaf/5.11.0.redhat-620133/xml/features). This format is deprecated and support for it will soon be removed
2015-08-31 23:32:50,499 | WARN  | nload-2-thread-3 | JaxbUtil                         | 83 - io.fabric8.fabric-agent - 1.2.0.redhat-133 | Old style feature file without namespace found (URI: mvn:org.apache.activemq/activemq-karaf/5.11.0.redhat-620133/xml/features-core). This format is deprecated and support for it will soon be removed
2015-08-31 23:32:50,565 | WARN  | nload-2-thread-1 | JaxbUtil                         | 83 - io.fabric8.fabric-agent - 1.2.0.redhat-133 | Old style feature file without namespace found (URI: mvn:org.apache.activemq/activemq-karaf/5.11.0.redhat-620133/xml/features-core). This format is deprecated and support for it will soon be removed
2015-08-31 23:32:52,804 | INFO  | a01c1-1-thread-1 | Agent                            | 83 - io.fabric8.fabric-agent - 1.2.0.redhat-133 | No deployment change.
2015-08-31 23:32:52,807 | INFO  | a01c1-1-thread-1 | Agent                            | 83 - io.fabric8.fabric-agent - 1.2.0.redhat-133 | Stopping bundles:
2015-08-31 23:32:52,808 | INFO  | a01c1-1-thread-1 | Agent                            | 83 - io.fabric8.fabric-agent - 1.2.0.redhat-133 |   org.ops4j.base.util.property / 1.5.0
2015-08-31 23:32:52,810 | INFO  | pool-20-thread-1 | Deployer                         | 83 - io.fabric8.fabric-agent - 1.2.0.redhat-133 | Scheduled stop for bundle:org.ops4j.base.util.property with a timeout limit of 60 seconds
2015-08-31 23:32:52,815 | INFO  | a01c1-1-thread-1 | Agent                            | 83 - io.fabric8.fabric-agent - 1.2.0.redhat-133 |   com.google.guava / 17.0.0
2015-08-31 23:32:52,815 | INFO  | pool-20-thread-1 | Deployer                         | 83 - io.fabric8.fabric-agent - 1.2.0.redhat-133 | Scheduled stop for bundle:com.google.guava with a timeout limit of 60 seconds
2015-08-31 23:32:52,816 | INFO  | a01c1-1-thread-1 | Agent                            | 83 - io.fabric8.fabric-agent - 1.2.0.redhat-133 |   org.apache.commons.codec / 1.10.0
2015-08-31 23:32:52,816 | INFO  | pool-20-thread-1 | Deployer                         | 83 - io.fabric8.fabric-agent - 1.2.0.redhat-133 | Scheduled stop for bundle:org.apache.commons.codec with a timeout limit of 60 seconds
2015-08-31 23:32:52,817 | INFO  | a01c1-1-thread-1 | Agent                            | 83 - io.fabric8.fabric-agent - 1.2.0.redhat-133 |   org.apache.servicemix.bundles.jasypt / 1.9.2.1
2015-08-31 23:32:52,817 | INFO  | pool-20-thread-1 | Deployer                         | 83 - io.fabric8.fabric-agent - 1.2.0.redhat-133 | Scheduled stop for bundle:org.apache.servicemix.bundles.jasypt with a timeout limit of 60 seconds
2015-08-31 23:32:52,817 | INFO  | a01c1-1-thread-1 | Agent                            | 83 - io.fabric8.fabric-agent - 1.2.0.redhat-133 |   org.apache.commons.lang / 2.6.0
2015-08-31 23:32:52,818 | INFO  | pool-20-thread-1 | Deployer                         | 83 - io.fabric8.fabric-agent - 1.2.0.redhat-133 | Scheduled stop for bundle:org.apache.commons.lang with a timeout limit of 60 seconds
2015-08-31 23:32:52,818 | INFO  | a01c1-1-thread-1 | Agent                            | 83 - io.fabric8.fabric-agent - 1.2.0.redhat-133 |   org.apache.commons.beanutils / 1.8.3
2015-08-31 23:32:52,819 | INFO  | pool-20-thread-1 | Deployer                         | 83 - io.fabric8.fabric-agent - 1.2.0.redhat-133 | Scheduled stop for bundle:org.apache.commons.beanutils with a timeout limit of 60 seconds
2015-08-31 23:32:52,819 | INFO  | a01c1-1-thread-1 | Agent                            | 83 - io.fabric8.fabric-agent - 1.2.0.redhat-133 |   org.apache.commons.collections / 3.2.1
2015-08-31 23:32:52,819 | INFO  | pool-20-thread-1 | Deployer                         | 83 - io.fabric8.fabric-agent - 1.2.0.redhat-133 | Scheduled stop for bundle:org.apache.commons.collections with a timeout limit of 60 seconds
2015-08-31 23:32:52,820 | INFO  | a01c1-1-thread-1 | Agent                            | 83 - io.fabric8.fabric-agent - 1.2.0.redhat-133 |   org.apache.servicemix.bundles.jsch / 0.1.51.1
2015-08-31 23:32:52,820 | INFO  | pool-20-thread-1 | Deployer                         | 83 - io.fabric8.fabric-agent - 1.2.0.redhat-133 | Scheduled stop for bundle:org.apache.servicemix.bundles.jsch with a timeout limit of 60 seconds
2015-08-31 23:32:52,820 | INFO  | a01c1-1-thread-1 | Agent                            | 83 - io.fabric8.fabric-agent - 1.2.0.redhat-133 |   org.mvel2 / 2.2.1.Final
2015-08-31 23:32:52,820 | INFO  | pool-20-thread-1 | Deployer                         | 83 - io.fabric8.fabric-agent - 1.2.0.redhat-133 | Scheduled stop for bundle:org.mvel2 with a timeout limit of 60 seconds
2015-08-31 23:32:52,870 | INFO  | a01c1-1-thread-1 | Agent                            | 83 - io.fabric8.fabric-agent - 1.2.0.redhat-133 | Done.
2015-08-31 23:33:11,904 | INFO  | pool-10-thread-1 | FabricCxfRegistrationHandler     | 98 - io.fabric8.fabric-cxf-registry - 1.2.0.redhat-133 | Registering endpoint io.fabric8.cxf:bus.id=io.fabric8.fabric-rest-cxf2103332261,type=Bus.Service.Endpoint,service="{http://rest.fabric8.io/}FabricResource",port="FabricResource",instance.id=712303913 type null has status STARTEDat /fabric8
2015-08-31 23:33:11,906 | INFO  | pool-10-thread-1 | FabricCxfRegistrationHandler     | 98 - io.fabric8.fabric-cxf-registry - 1.2.0.redhat-133 | Found CXF servlet path from config admin: /cxf
2015-08-31 23:33:12,773 | INFO  | pool-10-thread-1 | FabricCxfRegistrationHandler     | 98 - io.fabric8.fabric-cxf-registry - 1.2.0.redhat-133 | Registered CXF API at /fabric/registry/clusters/apis/rest/FabricResource/fabric8/1.0/bookingflightcon JSON: {"id":"bookingflightcon", "container":"bookingflightcon", "version":"1.0", "services":["${zk:bookingflightcon/http}/cxf/fabric8"], "objectName":"io.fabric8.cxf:bus.id=io.fabric8.fabric-rest-cxf2103332261,type=Bus.Service.Endpoint,service=\"{http://rest.fabric8.io/}FabricResource\",port=\"FabricResource\",instance.id=712303913", "wadl":"?_wadl", "apidocs":"/api-docs"}
2015-08-31 23:33:12,787 | INFO  | pool-10-thread-1 | FabricCxfRegistrationHandler     | 98 - io.fabric8.fabric-cxf-registry - 1.2.0.redhat-133 | Registering endpoint io.fabric8.cxf:bus.id=io.fabric8.fabric-rest-cxf2103332261,type=Bus.Service.Endpoint,service="{http://rest.fabric8.io/}FabricResource",port="FabricResource",instance.id=712303913 type null has status STARTEDat /fabric8
2015-08-31 23:33:12,788 | INFO  | pool-10-thread-1 | FabricCxfRegistrationHandler     | 98 - io.fabric8.fabric-cxf-registry - 1.2.0.redhat-133 | Found CXF servlet path from config admin: /cxf
2015-08-31 23:33:12,793 | INFO  | pool-10-thread-1 | FabricCxfRegistrationHandler     | 98 - io.fabric8.fabric-cxf-registry - 1.2.0.redhat-133 | Registered CXF API at /fabric/registry/clusters/apis/rest/FabricResource/fabric8/1.0/bookingflightcon JSON: {"id":"bookingflightcon", "container":"bookingflightcon", "version":"1.0", "services":["${zk:bookingflightcon/http}/cxf/fabric8"], "objectName":"io.fabric8.cxf:bus.id=io.fabric8.fabric-rest-cxf2103332261,type=Bus.Service.Endpoint,service=\"{http://rest.fabric8.io/}FabricResource\",port=\"FabricResource\",instance.id=712303913", "wadl":"?_wadl", "apidocs":"/api-docs"}
2015-08-31 23:33:12,981 | INFO  | pool-10-thread-1 | HttpServiceFactoryImpl           | 193 - org.ops4j.pax.web.pax-web-runtime - 3.2.3 | Binding bundle: [io.fabric8.fabric-git-server [99]] to http service
2015-08-31 23:33:13,053 | INFO  | admin-1-thread-1 | FabricConfigAdminBridge          | 82 - io.fabric8.fabric-configadmin - 1.2.0.redhat-133 | Updating configuration org.ops4j.pax.logging
2015-08-31 23:33:13,058 | INFO  | admin-1-thread-1 | FabricConfigAdminBridge          | 82 - io.fabric8.fabric-configadmin - 1.2.0.redhat-133 | Updating configuration jmx.acl
```

Maybe this log entry can tell something:

```
2015-08-31 23:32:21,861 | INFO  | FelixStartLevel  | BlueprintContainerImpl           | 14 - org.apache.aries.blueprint.core - 1.4.2 | Bundle travelagency is waiting for namespace handlers [http://aries.apache.org/xmlns/jpa/v1.0.0]
```

**Comment by weimeilin79** [comment]
Yep, that was what I am looking for. See if you have these feature in your profile? 
![screen shot 2015-09-01 at 8 36 24](https://cloud.githubusercontent.com/assets/2534483/9604300/c3bab362-5084-11e5-9691-97511dd1a3da.png)

**Comment by weimeilin79** [comment]
I found there are some problems when merging, so please try to clone it again!

**Comment by rafaeltuelho** [comment]
Yeah!

Now the containers are running in a green state :-) See
![Containers state](https://rafaeltuelho.files.wordpress.com/2015/09/fuse62-jbossdemo-travel-agency.png)

NOTE: I had to build the fuse projects and after that I had to open the profiles and hit a `refresh` operation to force Fuse update the artifacts. I think it's because the artifact version is the same and are not using the `SNAPSHOT` qualifier.
